src/blog/2024/06/calling-python-script-from-node-red.md

```diff
@@ -0,0 +1,112 @@
+{% endrenderFlow %}
+
+Now, let's execute this Python script from Node-RED. To do that, we will use Node-RED's [Exec](/node-red/core-nodes/exec/) node, which allows running commands on your system.
+
+1. Drag an Inject node onto the canvas.
+2. Drag an Exec node onto the canvas and Configure the command to `python ./example.py -u`. The -u flag prevents potential output buffering issues when executing Python scripts via exec.
+
+!["Screenshot of the Exec node executing python file"](./images/calling-python-scrpt-from-node-red-exec-node.png "Screenshot of the Exec node executing python file"){data-zoomable}
+
+3. Drag an Debug node onto the canvas.
+4. Connect the output of the Inject node to input of Exec node, and output of Exec node to input of Debug node.
+
+{% renderFlow %}
+[{"id":"2e26b84c0ce17312","type":"exec","z":"FFF0000000000001","command":"python ./example.py -u","addpay":"","append":"","useSpawn":"false","timer":"","winHide":false,"oldrc":false,"name":"","x":460,"y":300,"wires":[["89589c56117004e0"],["7fdb901b144749c2"],["3f49b49308941782"]]},{"id":"739e08c1ec77c2a1","type":"inject","z":"FFF0000000000001","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":240,"y":300,"wires":[["2e26b84c0ce17312"]]},{"id":"89589c56117004e0","type":"debug","z":"FFF0000000000001","name":"Output","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":690,"y":260,"wires":[]},{"id":"7fdb901b144749c2","type":"debug","z":"FFF0000000000001","name":"Error","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":690,"y":300,"wires":[]},{"id":"3f49b49308941782","type":"debug","z":"FFF0000000000001","name":"Return code","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","targetType":"msg","statusVal":"","statusType":"auto","x":710,"y":340,"wires":[]}]
+{% endrenderFlow %}
+
+Now, when you deploy this flow and click on the inject node to execute the file, you should see the text 'Positive number entered' and `{ code: 0 }`, which indicates your script has been successfully executed.
+
+## Reading Temperature Sensor using Python script
+
+Now that we've covered how to execute a Python script from Node-RED with a simple example, let's dive into a more practical demonstration where we read sensor data using Python. While Node-RED has a wide range of community nodes that allow reading different sensors, for better demonstration this example is great so we will use Python for this purpose.
```

Comment:
```suggestion
Having explored how to run a Python script within Node-RED, let's move from theory to practice with a real-world example. We'll demonstrate how to read sensor data using Python, even though Node-RED itself offers numerous community-built nodes for this purpose. This approach offers deeper insights into integrating external scripts, showcasing the flexibility of Node-RED for custom solutions.
```

Reply:
the above example is also practical, so need to remove theory word